build.cmd

```diff
@@ -1,3 +1,34 @@
-@echo off
+@echo Off
 
-call bundle exec rake -f rakefile.rb %*
+rem change working directory to the repository root
+pushd %~dp0
+setlocal
+
+rem options
+set NUGET_VERSION=3.5.0
+
+rem determine nuget cache dir
+set NUGET_CACHE_DIR=%LocalAppData%\.nuget\v%NUGET_VERSION%
+
+rem download nuget to cache dir
+set NUGET_URL=https://dist.nuget.org/win-x86-commandline/v%NUGET_VERSION%/NuGet.exe
+if not exist %NUGET_CACHE_DIR%\NuGet.exe (
+  if not exist %NUGET_CACHE_DIR% md %NUGET_CACHE_DIR%
+  echo Downloading '%NUGET_URL%'' to '%NUGET_CACHE_DIR%\NuGet.exe'...
+  @powershell -NoProfile -ExecutionPolicy unrestricted -Command "$ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest '%NUGET_URL%' -OutFile '%NUGET_CACHE_DIR%\NuGet.exe'"
+)
+
+rem copy nuget locally
+if not exist .nuget\NuGet.exe (
+  if not exist .nuget md .nuget
+  copy %NUGET_CACHE_DIR%\NuGet.exe .nuget\NuGet.exe > nul
+)
+
+rem restore packages for build script
+.nuget\NuGet.exe restore .nuget\packages.config -PackagesDirectory .\packages -Verbosity quiet

```

Comment:
Another option is to remove the `static` modifier from the methods; then we don't need to declare any variable `static`. IMO this is better than passing everything as a parameter. This is C#, sure, but it's also a script, and having to pass parameters around all the time doesn't feel very "scripty".